src/hidimstat/conditional_sampling.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+from sklearn.base import MultiOutputMixin, check_is_fitted
+from sklearn.multioutput import MultiOutputClassifier, MultiOutputRegressor
+from sklearn.utils.validation import check_random_state
+
+
+class ConditionalSampler:
+    def __init__(
+        self,
+        model_regression=None,
+        model_binary=None,
+        model_categorical=None,
+        data_type: str = "auto",
+        random_state=None,
+        categorical_max_cardinality=10,
+    ):
+        """
+        Class use to sample from the conditional distribution $p(X^j | X^{-j})$.
+
+        Parameters
+        ----------
+        model_regression : object
+            The model to use for continuous data.
+        model_binary : object
```

Comment:
same here and below

Reply:
Bzw binary is just a special case of categorical. I propose not to handle it explicitly.

Reply:
Agreed, I replaced it.